File: dotNET/pdfclown.samples.cli/src/org/pdfclown/samples/cli/AcroFormFillingSample.cs
using org.pdfclown.documents;
using org.pdfclown.documents.contents.composition;
using org.pdfclown.documents.contents.entities;
using org.pdfclown.documents.contents.fonts;
using org.pdfclown.documents.contents.xObjects;
using org.pdfclown.documents.interaction.annotations;
using org.pdfclown.documents.interaction.forms;
using org.pdfclown.files;

using System;
using System.Collections.Generic;
using System.Drawing;

namespace org.pdfclown.samples.cli
{
  /**
    <summary>This sample demonstrates how to fill AcroForm fields of a PDF document.</summary>
  */
  public class AcroFormFillingSample
    : Sample
  {
    public override bool Run(
      )
    {
      string filePath = PromptPdfFileChoice("Please select a PDF file");

      // 1. Open the PDF file!
      File file = new File(filePath);
      Document document = file.Document;

      // 2. Get the acroform!
      Form form = document.Form;
      if(form == null)
      {Console.WriteLine("\nNo acroform available (AcroForm dictionary not found).");}
      else
      {
        Console.WriteLine("\nPlease insert a value for each field listed below (or type 'quit' to end this sample).\n");

        // 3. Filling the acroform fields...
        foreach(Field field in form.Fields.Values)
        {
          Console.WriteLine("* " + field.GetType().Name + " '" + field.FullName + "' (" + field.BaseObject + "): ");
          Console.WriteLine("    Current Value:" + field.Value);
          string newValue = PromptChoice("    New Value:");
          if(newValue != null && newValue.Equals("quit"))
            break;

          field.Value = newValue;
        }
      }

      // 4. Serialize the PDF file!
      Serialize(file,false);

      return true;
    }
  }
}

File: java/pdfclown.samples.cli/src/org/pdfclown/samples/cli/AcroFormFillingSample.java
package org.pdfclown.samples.cli;

import org.pdfclown.documents.Document;
import org.pdfclown.documents.interaction.forms.Field;
import org.pdfclown.documents.interaction.forms.Form;
import org.pdfclown.files.File;

/**
  This sample demonstrates <b>how to fill AcroForm fields</b> of a PDF document.

  @author Stefano Chizzolini (http://www.stefanochizzolini.it)
  @since 0.1.1
  @version 0.1.1, 07/05/11
*/
public class AcroFormFillingSample
  extends Sample
{
  @Override
  public boolean run(
    )
  {
    String filePath = promptPdfFileChoice("Please select a PDF file");

    // 1. Open the PDF file!
    File file;
    try
    {file = new File(filePath);}
    catch(Exception e)
    {throw new RuntimeException(filePath + " file access error.",e);}

    Document document = file.getDocument();

    // 2. Get the acroform!
    Form form = document.getForm();
    if(form == null)
    {System.out.println("\nNo acroform available (AcroForm dictionary not found).");}
    else
    {
      System.out.println("\nPlease insert a value for each field listed below (or type 'quit' to end this sample).\n");

      // 3. Filling the acroform fields...
      for(Field field : form.getFields().values())
      {
        System.out.println("* " + field.getClass().getSimpleName() + " '" + field.getFullName() + "' (" + field.getBaseObject() + "): ");
        System.out.println("    Current Value:" + field.getValue());
        String newValue = promptChoice("    New Value:");
        if(newValue != null && newValue.equals("quit"))
          break;

        field.setValue(newValue);
      }
    }

    // 4. Serialize the PDF file!
    serialize(file,false);

    return true;
  }
}
